test/releasers/node.ts

```diff
@@ -0,0 +1,201 @@
+  [skip: string]: Function;
+}
+function requireNode10(this: MochaThis) {
+  const match = process.version.match(/v([0-9]+)/);
+  if (match) {
+    if (Number(match[1]) < 10) this.skip();
+  }
+}
+function mockRequest(snapName: string) {
+  const packageContent = readFileSync(
+    resolve(fixturesPath, 'package.json'),
+    'utf8'
+  );
+  const graphql = JSON.parse(
+    readFileSync(resolve(fixturesPath, 'commits.json'), 'utf8')
+  );
+
+  const req = nock('https://api.github.com')
+    .get('/repos/googleapis/node-test-repo/pulls?state=closed&per_page=100')
+    .reply(200, undefined)
+    .get('/repos/googleapis/node-test-repo/contents/package.json')
+    .reply(200, {
+      content: Buffer.from(packageContent, 'utf8').toString('base64'),
+      sha: 'abc123',
+    })
+    // fetch semver tags, this will be used to determine
+    // the delta since the last release.
+    .get('/repos/googleapis/node-test-repo/tags?per_page=100')
+    .reply(200, [
+      {
+        name: 'v0.123.4',
+        commit: {
+          sha: 'da6e52d956c1e35d19e75e0f2fdba439739ba364',
+        },
+      },
+    ])
+    .post('/graphql')
+    .reply(200, {
+      data: graphql,
+    })
+    // getting the latest tag
+    .get('/repos/googleapis/node-test-repo/git/refs?per_page=100')
+    .reply(200, [{ref: 'refs/tags/v0.123.4'}])
+    // creating a new branch
+    .post('/repos/googleapis/node-test-repo/git/refs')
+    .reply(200)
+    // check for CHANGELOG
+    .get(
+      '/repos/googleapis/node-test-repo/contents/CHANGELOG.md?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
+    .put(
+      '/repos/googleapis/node-test-repo/contents/CHANGELOG.md',
+      (req: {[key: string]: string}) => {
+        snapshot(`CHANGELOG-node-message-${snapName}`, req.message);
+        snapshot(
+          `CHANGELOG-node-${snapName}`,
+          Buffer.from(req.content, 'base64')
+            .toString('utf8')
+            .replace(/\([0-9]{4}-[0-9]{2}-[0-9]{2}\)/g, '')
+        );
+        return true;
+      }
+    )
+    .reply(201)
+    // update package.json
+    .get(
+      '/repos/googleapis/node-test-repo/contents/package.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(200, {
+      content: Buffer.from(packageContent, 'utf8').toString('base64'),
+      sha: 'abc123',
+    })
+    .put(
+      '/repos/googleapis/node-test-repo/contents/package.json',
+      (req: {[key: string]: string}) => {
+        snapshot(`package-json-node-message-${snapName}`, req.message);
+        snapshot(
+          `package-json-node-${snapName}`,
+          Buffer.from(req.content, 'base64').toString('utf8')
+        );
+        return true;
+      }
+    )
+    .reply(200)
+    .get(
+      '/repos/googleapis/node-test-repo/contents/samples/package.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
+    .get(
+      '/repos/googleapis/node-test-repo/contents/samples/package-lock.json?ref=refs%2Fheads%2Frelease-v0.123.5'
+    )
+    .reply(404)
```

Comment:
I have a PR I expect to land early in the week which adds support for alternate default branch names on GitHub:

https://github.com/googleapis/release-please/pull/468

We'll need to make a small update to this test once that lands.

Reply:
No problem. I can fix once it is applied